### ISSUE TITLE
Thumbnails 

### ISSUE THREAD

**Issue by Bouni** [reaktor23/website#12]
We shoud look for a way to generate thumbnails of images that are clickable and then provide a high res of the picture to get better loading times.
Maybe there's a ready to use solution available.

**Comment by Bouni** [comment]
🤔 
https://gohugo.io/content-management/image-processing/

**Comment by samuelb** [comment]
We use it already in the project overview: https://github.com/reaktor23/website/blob/4b6c76ebdc26ec8dcf849714b9188dc146831730/themes/reaktor23/layouts/projects/list.html#L10

Where do you want to have those thumbnails?

**Comment by Bouni** [comment]
In all the project pages. I would like to have thumbnails in the project pages that are clickable to get a large view.

**Comment by samuelb** [comment]
As we discussed yesterday, I continued to modify the `thumbnail` shortcode to make use of [hugos image resize capability](https://gohugo.io/content-management/image-processing/) in 0da48e87e5c5910a468f7753ef3895dfd45008c6. With 91f7f2a5aaa2c8e3e4343a965f42d3960981f95b the images are also linked to the original ones without much fanciness.

**Comment by Bouni** [comment]
Nice, maybe we could also integrate some sort of JS image preview modal thingy that allows to view the large version without loading the image in a new page. Just for comfort

**Comment by samuelb** [comment]
Erm, let's follow up here.

Back then we extended the [thumbnail shortcode](https://github.com/reaktor23/website/blob/master/themes/reaktor23/layouts/shortcodes/thumbnail.html). What would be nice for this is an optional caption text below the image. Also, we should make possible to have multiple thumbnails in one row while still being properly aligned. The current blog article with multiple pictures underneath looks a bit ugly. I started to play around recently already in the branch [thumbnails-captions](https://github.com/reaktor23/website/tree/thumbnails-captions).

And then I stumbled over hugo's build-in shortcode [{{< figure ... >}}](https://gohugo.io/content-management/shortcodes/#figure) which has somehow similar what not the same purpose as we build with the thumbnail shortcode. Currently, there is no CSS to render figure nicely. We may could use both, one for to create scale images, the other for non-scaled, but both displayed similar.
